### DISCUSSION TITLE
Blocking moving windows by mouse

### DISCUSSION THREAD

**Discussion by MattOrmianek** [rxhanson/RectanglePro-Community#571] [Ideas]
Hi, I frequently use Rectangle Pro and have noticed some odd behavior with my mouse. It's not an issue with the mouse itself, but more likely how I’m using it. To the point, I think having an option to block window movement would be really helpful for me, and possibly for others as well.

**Comment by rxhanson** [comment]
Thanks for the feedback. I'm not quite sure I understand what's going wrong. Maybe a quick screen recording would help me understand better?

**Reply by MattOrmianek** [reply]
Hi, this is feature request, not bug. Due to the unconventional positioning of my trackpad and mouse, I occasionally click and slightly move windows unintentionally. Some applications, like the Arc Browser, have large panels that make it easy to inadvertently drag and reposition the window, and there isn't an option within these apps to disable this behavior.

I'm looking for a way to block the possibility of moving windows to prevent this issue. After spending some time working, I often end up with windows overlapping each other by 50-100 pixels, which disrupts my workflow. Implementing a feature to lock window positions would greatly enhance accessibility for users like me who face similar challenges.

I've attached a screenshot showing the typical result after a few minutes of work.
![CleanShot 2024-10-11 at 3  00 44](https://github.com/user-attachments/assets/14d7d4e8-1c35-4436-9f89-1d35d1dfa577)

**Reply by MattOrmianek** [reply]
Here is another screenshot showing exactly what is going on
![CleanShot 2024-10-11 at 3  05 26](https://github.com/user-attachments/assets/6f481b09-d570-435c-9b45-c3acebbd2189)

**Reply by rxhanson** [reply]
I see. Although not exactly what you're looking for, you could create App Layout or Custom Size & Position configurations, and in those you can check the box for "Use as a snap target". With this configured, any time you drag a window within that area then it would snap it back into that size/position. If you did this in Custom Size & Position, then it would work globally for all apps, whereas if you made App Layouts then you could constrain it to specific apps. 

Perhaps it would make sense for me to add functionality that allows snap targets to be disregarded when a modifier key is held. The opposite behavior (hold a modifier to activate snap targets) is enabled in the Snap Areas tab in the Settings window.

The downside of Snap Targets in your specific scenario is that you could accidentally snap to the wrong target, but maybe that will be uncommon enough for it to still solve the issue? Let me know if you can't figure out the Snap Targets since it has a higher learning curve than other features in the app.

**Reply by MattOrmianek** [reply]
Okay, so I followed your instructions, and the configuration is shown in the screenshot. Unfortunately, it does not resolve the issue. I use shortcuts to move windows into place, but I can still move them with the mouse, and after a while, they are not in their correct positions. While I can use a shortcut to move them back, that's not the point.
![CleanShot 2024-10-11 at 7  15 00@2x](https://github.com/user-attachments/assets/e5d4f713-1e93-4d12-98e6-68dc80c40826)

I believe that saving the positions of windows when an option like "disable moving windows by mouse" is enabled would be a simple solution. Yabai uses something similar. One option is to block moving windows with the mouse (using some Apple-specific method), and another is to save the positions of windows (updating when windows are moved by keyboard shortcuts). I don't want to sound difficult.

**Reply by rxhanson** [reply]
Thanks for trying it out and posting back. It seems clear that this is not what you're looking for, but the reason to consider Snap Targets was to make it so that you wouldn't have to execute a shortcut to move a window back. The snap target would take effect when you release the window, automatically snapping it back into place (this is a feature similar to how fancy zones works on Windows Power Toys). If that didn't work in your testing, then perhaps there's a bug.

If there's enough demand for a setting to disable moving windows using the mouse, I'll consider it.
